From 73af84391c89f486180f307d7bf8bf549aec43ea Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 8 Mar 2018 15:35:00 -0800
Subject: [PATCH] TRITON-218 drop SMF manifest templating in sapi Reviewed by:
 Dylan Yep <dyep49@gmail.com> Approved by: Dylan Yep <dyep49@gmail.com>

---
 Makefile                                | 2 +-
 boot/configure.sh                       | 8 +-------
 boot/setup.sh                           | 4 +++-
 smf/manifests/{sapi.xml.in => sapi.xml} | 4 ++--
 4 files changed, 7 insertions(+), 11 deletions(-)
 rename smf/manifests/{sapi.xml.in => sapi.xml} (81%)

diff --git a/Makefile b/Makefile
index 5a37add..cc5b416 100644
--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,7 @@ JSL_CONF_NODE	 = tools/jsl.node.conf
 JSL_FILES_NODE   = $(JS_FILES)
 JSSTYLE_FILES	 = $(JS_FILES)
 JSSTYLE_FLAGS    = -o indent=4,doxygen,unparenthesized-return=0
-SMF_MANIFESTS_IN = smf/manifests/sapi.xml.in
+SMF_MANIFESTS	 = smf/manifests/sapi.xml
 
 NODE_PREBUILT_VERSION=v4.8.7
 ifeq ($(shell uname -s),SunOS)
diff --git a/boot/configure.sh b/boot/configure.sh
index 1cb5f98..8f075b6 100755
--- a/boot/configure.sh
+++ b/boot/configure.sh
@@ -7,16 +7,10 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 set -o xtrace
 
-echo "Updating SMF manifest"
-$(/opt/local/bin/gsed -i"" -e "s/@@PREFIX@@/\/opt\/smartdc\/sapi/g" /opt/smartdc/sapi/smf/manifests/sapi.xml)
-
-echo "Importing sapi.xml"
-/usr/sbin/svccfg import /opt/smartdc/sapi/smf/manifests/sapi.xml
-
 exit 0
diff --git a/boot/setup.sh b/boot/setup.sh
index a6556c9..007c577 100755
--- a/boot/setup.sh
+++ b/boot/setup.sh
@@ -7,7 +7,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
@@ -109,6 +109,8 @@ HERE
 
 fi
 
+/usr/sbin/svccfg import /opt/smartdc/sapi/smf/manifests/sapi.xml
+
 echo "Adding log rotation"
 sdc_log_rotation_add amon-agent /var/svc/log/*amon-agent*.log 1g
 sdc_log_rotation_add config-agent /var/svc/log/*config-agent*.log 1g
diff --git a/smf/manifests/sapi.xml.in b/smf/manifests/sapi.xml
similarity index 81%
rename from smf/manifests/sapi.xml.in
rename to smf/manifests/sapi.xml
index 1129959..06a43d4 100644
--- a/smf/manifests/sapi.xml.in
+++ b/smf/manifests/sapi.xml
@@ -7,7 +7,7 @@
 -->
 
 <!--
-    Copyright (c) 2014, Joyent, Inc.
+    Copyright (c) 2018, Joyent, Inc.
 -->
 
 <service_bundle type="manifest" name="smartdc-application-sapi">
@@ -20,7 +20,7 @@
 	<service_fmri value="svc:/system/filesystem/local" />
 </dependency>
 
-<exec_method type="method" name="start" exec="@@NODE@@ --abort-on-uncaught-exception @@PREFIX@@/server.js -f @@PREFIX@@/etc/config.json &amp;" timeout_seconds="30" />
+<exec_method type="method" name="start" exec="/opt/smartdc/sapi/build/node/bin/node --abort-on-uncaught-exception /opt/smartdc/sapi/server.js -f /opt/smartdc/sapi/etc/config.json &amp;" timeout_seconds="30" />
 <exec_method type="method" name="stop" exec=":kill" timeout_seconds="30" />
 
 <instance name="default" enabled="true" />
-- 
2.21.0

